persistent-test/src/NullableGenerated.hs

```diff
@@ -0,0 +1,20 @@
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TypeApplications #-}
+
+-- Used in PersistLiteralTestSQL
+module NullableGenerated where
+
+import qualified Data.Proxy as P (Proxy(..))
+import Init
+
+newtype NullableGenerated a = NullableGenerated (Maybe a) deriving (Show, Eq)
+
+instance forall a. PersistField a => PersistField (NullableGenerated a) where
+  toPersistValue (NullableGenerated _) = PersistLiteral "DEFAULT"
+  fromPersistValue g = NullableGenerated <$> fromPersistValue g
```

Comment:
This is a useful pattern for describing how this works but I really can't recommend using it in app code. Having it present in the tests is a bit worrying to me, as well, as other people might mimic it and then come across bugs for it.